src/vehicle_use.cpp

```diff
@@ -1744,6 +1744,152 @@ void vehicle::use_harness( int part, const tripoint &pos )
     }
 }
 
+std::vector<item_count_tuple> vehicle::get_comestible_liquids()
+{
+  std::vector<item_count_tuple> liquids;
+  std::map<item*, uint32_t> liquids_map;
+
+  liquids.clear();
+  liquids_map.clear();
+
+  for ( auto &part : parts)
+  {
+    const auto remaining = part.ammo_remaining();
+
+    if (part.is_tank() && !part.base.contents_made_of( SOLID ) && remaining > 0)
+    {
+      const auto food = part.base.get_food();
+
+      if (food != nullptr && food->is_comestible())
+      {
+        const auto comestible = food->get_comestible();
+
+        if (comestible != nullptr && "DRINK" == comestible->comesttype)
+        {
+          const auto liquid_id = part.ammo_current();
+
+          if (!liquid_id.empty() && liquid_id != "null" && liquid_id != "water")
+          {
+            if (liquids_map.find(food) == liquids_map.end())
+            {
+              liquids_map[food] = static_cast<uint32_t>(remaining);
+            }
+            else
+            {
+              liquids_map[food] += static_cast<uint32_t>(remaining);
```

Comment:
Tip: you can use `continue` in loops to avoid large indentations like this:
```C++
for( ... ) {
    if( !condition0 ) {
        continue;
    }
    if( !condition1 ) {
        continue;
    }
    if( !condition2 ) {
        continue;
    }
    // remaining code
}
```

The remaining code is only intended once.

Reply:
You are right, sir. Changed it. THX for hinting at this.